cardano-api/src/Cardano/Api/Convenience/Query.hs

```diff
@@ -63,48 +66,37 @@ queryStateForBalancedTx
   -> NetworkId
   -> [TxIn]
   -> IO (Either QueryConvenienceError (UTxO era, ProtocolParameters, EraHistory CardanoMode, SystemStart, Set PoolId))
-queryStateForBalancedTx era networkId allTxIns = do
- eSocketPath <- first SockErr <$> readEnvSocketPath
- case eSocketPath of
-  Left e -> return $ Left e
-  Right (SocketPath sockPath) -> do
-    let cModeParams = CardanoModeParams $ EpochSlots 21600
-        localNodeConnInfo = LocalNodeConnectInfo
-                              cModeParams
-                              networkId
-                              sockPath
-    eSbe <- return . getSbe $ cardanoEraStyle era
-    case eSbe of
-      Left e -> return $ Left e
-      Right qSbe -> do
-        case toEraInMode era CardanoMode of
-          Just qeInMode -> do
-
-            -- Queries
-            let utxoQuery = QueryInEra qeInMode $ QueryInShelleyBasedEra qSbe
-                              $ QueryUTxO (QueryUTxOByTxIn (Set.fromList allTxIns))
-                pparamsQuery = QueryInEra qeInMode
-                                 $ QueryInShelleyBasedEra qSbe QueryProtocolParameters
-                eraHistoryQuery = QueryEraHistory CardanoModeIsMultiEra
-                systemStartQuery = QuerySystemStart
-                stakePoolsQuery = QueryInEra qeInMode . QueryInShelleyBasedEra qSbe $ QueryStakePools
-
-            -- Query execution
-            eUtxo <- executeQueryCardanoMode era networkId utxoQuery
-            ePparams <- executeQueryCardanoMode era networkId pparamsQuery
-            eEraHistory <- queryNodeLocalState localNodeConnInfo Nothing eraHistoryQuery
-            eSystemStart <- queryNodeLocalState localNodeConnInfo Nothing systemStartQuery
-            eStakePools <- executeQueryCardanoMode era networkId stakePoolsQuery
-            return $ do
-              utxo <- eUtxo
-              pparams <- ePparams
-              eraHistory <- first AcqFailure eEraHistory
-              systemStart <- first AcqFailure eSystemStart
-              stakePools <- eStakePools
-              Right (utxo, pparams, eraHistory, systemStart, stakePools)
-          Nothing -> return $ Left $ EraConsensusModeMismatch
-                                      (AnyConsensusMode CardanoMode)
-                                      (getIsCardanoEraConstraint era $ AnyCardanoEra era)
+queryStateForBalancedTx era networkId allTxIns = runExceptT $ do
+  SocketPath sockPath <- ExceptT $ first SockErr <$> readEnvSocketPath
```

Comment:
I'm assuming you're not using `newExceptT` because `ExceptT` is shorter?

Reply:
Yeah, that's right.